crowd_anki/importer/anki_importer.py

```diff
@@ -62,21 +54,59 @@ def load_from_directory(self, directory_path, import_media=True):
     def get_deck_path(self, directory_path):
         """
         Provides compatibility layer between deck file naming conventions.
+        Assumes that deck json file is located in the directory and named 'deck.json'
         """
 
         def path_for_name(name):
             return directory_path.joinpath(name).with_suffix(DECK_FILE_EXTENSION)
 
-        convention_path = path_for_name(self.deck_file_name)
-        inferred_path = path_for_name(directory_path.name)
+        convention_path = path_for_name(self.deck_file_name)   # [folder]/deck.json
+        inferred_path = path_for_name(directory_path.name)     # [folder]/[folder].json
         return convention_path if convention_path.exists() else inferred_path
 
+    def load_deck_with_settings(self, directory_path) -> (dict, ImportConfig):
+        deck_json = self.read_deck(self.get_deck_path(directory_path))
+        import_config = self.read_import_config(directory_path)
+
+        import_dialog = ImportDialog(deck_json, import_config)
+        if import_dialog.exec_() == QDialog.Rejected:
+            return None, None  # User has cancelled
+
+        # TODO: strip settings from deck_json
+
+        return deck_json, import_dialog.final_import_config
+
     @staticmethod
-    def import_deck_from_path(collection, directory_path, import_media=True):
+    def read_deck(file_path):
+        """
+        Load deck from json file
+        :type file_path: Path
```

Comment:
just use the proper python type annotations instead

Reply:
I swear I copied this from somewhere else in your code :thinking: but I can no longer find it. Anyways, changed :sweat_smile: